### PR TITLE
Instagram-challenge (not styled, associations not great)

### PR DESCRIPTION
**Pull request by jackhalltipping**

### CODE REVIEW

**Comment by jackhalltipping** [comment]
Can we go over:
-best process for doing associations (like the order you do things in).
-why do I have too many leave a comment buttons?
-is there a guide anywhere for which files you create by hand and which you rails generate? or do you just learn that

**Comment by jackhalltipping** [comment]
Would also like to figure out how to pass my likes test. Getting this error
```Failure/Error: Unable to find matching line from backtrace
     ActionController::ParameterMissing:
       param is missing or the value is empty: picture```

Not sure how to go about fixing it.

File: spec/features/comments_feature_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+feature 'comments' do
```
Comment (ptolemybarnes) [inline spec/features/comments_feature_spec.rb]: There is a lot of repetition across all of these tests.

File: spec/features/users_feature_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+feature "User can sign in and out" do
+  context "user not signed in and on the homepage" do
+    it "should see a 'sign in' link and a 'sign up' link" do
+      visit('/')
```
Comment (ptolemybarnes) [inline spec/features/users_feature_spec.rb]: `visit('/')` repeated a lot

File: spec/features/users_feature_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    it "should not see 'sign out' link" do
+      visit('/')
+      expect(page).not_to have_link('Sign out')
+    end
+  end
+
+  context "user signed in on the homepage" do
+    before do
+      visit('/')
+      click_link('Sign up')
```
Comment (ptolemybarnes) [inline spec/features/users_feature_spec.rb]: Check out this gist for writing good helper methods of this kind: https://gist.github.com/ptolemybarnes/2dfda607b85d01e113b0

**Comment by ptolemybarnes** [comment]
- One of your controllers, the LikesController, serves JSON rather than HTML. This means that it's not a controller that serves user-facing content. It might be nice for the path to be different, as it is with GitHub (we go to api.github.com to access the API). Go [here](https://teamtreehouse.com/library/build-a-rails-api/namespacing-routes-and-json/namespaces-and-routes) for some pointers on that.
- Work on refactoring your tests.

Otherwise, I'm pretty happy with this. Nice one! :checkered_flag:

**Pull request closed without merging**